### PR TITLE
Feature/edit playlist picture

### PR DESCRIPTION
**Pull request by sandroquist**
This pull request introduces several changes to the playlist and profile management features, focusing on adding support for uploading and displaying playlist cover images. The most important changes include updating the `Playlist` data class, adding methods to handle cover images in the `PlaylistRepository`, and modifying the UI components to display the cover images.

### Playlist and Cover Image Management:

* [`app/src/main/java/com/epfl/beatlink/model/library/Playlist.kt`](diffhunk://#diff-88a7d187f15c4e2e86d304fc9867f05d8a4465b8ba75ad72e8533c712ccf88c8L21-R21): Updated the `Playlist` data class to allow `playlistCover` to be nullable.
* [`app/src/main/java/com/epfl/beatlink/model/library/PlaylistRepository.kt`](diffhunk://#diff-bc9b74e32b0dec328886047df8a77405f8a835feb7784eb007e7f26d778b01b5R57-R62): Added methods to upload and load playlist cover images.
* [`app/src/main/java/com/epfl/beatlink/repository/library/PlaylistRepositoryFirestore.kt`](diffhunk://#diff-0ca73801d9811ac319335b09f789a22ba8bcc4da64ffb25fec41f9f42903d736R315-R349): Implemented the methods to handle cover image upload and retrieval in Firestore.

### UI Components:

* [`app/src/main/java/com/epfl/beatlink/ui/components/library/PlaylistCard.kt`](diffhunk://#diff-b75d3c4f4e8dc6708a812781af3c0a0bda62ab72adde79945b599f30102eca01R40-R42): Modified the `PlaylistCard` component to load and display the playlist cover image.
* [`app/src/main/java/com/epfl/beatlink/ui/components/library/PlaylistCover.kt`](diffhunk://#diff-b36e950f43b9aea5313fb7a9646c566b074dabd566d8346e555b08d4022767c6L3-R22): Created a new `PlaylistCover` composable to handle the display of cover images.

### Profile Management:

* [`app/src/main/java/com/epfl/beatlink/repository/profile/ProfileRepositoryFirestore.kt`](diffhunk://#diff-4b084ab579838f2c552fa2f1943c2de3b8eeaae09c8e3794d6c0ba07de5e174bL217-L267): Removed redundant methods for image resizing and Base64 conversion, now using utility functions.

### CODE REVIEW

File: app/src/main/java/com/epfl/beatlink/ui/library/CreateNewPlaylist.kt [hunk 1]
```diff
@@ -165,6 +189,9 @@ fun CreateNewPlaylistScreen(
                           playlistTracks = emptyList(),
                           nbTracks = 0)
                   playlistViewModel.addPlaylist(newPlaylist)
+                  if (imageUri != null) {
```
Comment (lunasrc) [inline app/src/main/java/com/epfl/beatlink/ui/library/CreateNewPlaylist.kt:192]: I think here should be `imageUri != Uri.EMPTY` as the default value is set to `Uri.EMPTY`

File: app/src/main/java/com/epfl/beatlink/ui/library/EditPlaylist.kt [hunk 2]
```diff
@@ -193,6 +222,9 @@ fun EditPlaylistScreen(
                           nbTracks = selectedPlaylistState.nbTracks)
                   playlistViewModel.updatePlaylist(updatedPlaylist)
                   playlistViewModel.selectPlaylist(updatedPlaylist)
+                  if (imageUri != Uri.EMPTY) {
+                    playlistViewModel.uploadPlaylistCover(imageUri, context, updatedPlaylist)
+                  }
```
Comment (lunasrc) [inline app/src/main/java/com/epfl/beatlink/ui/library/EditPlaylist.kt:227]: Here if we modify the cover image and put a new image, is the new one uploaded ?

**Review by lunasrc — CHANGES_REQUESTED** [review]
Overall, the PR adds useful functionality for creating and editing playlists, and it handles playlist cover images effectively. With a few minor adjustments in code comments for clarity and one clarification to confirm, this PR will be ready for final approval. Keep up the great work! :)

File: app/src/main/java/com/epfl/beatlink/ui/library/EditPlaylist.kt (hunk 2 above)
Comment (sandroquist) [reply app/src/main/java/com/epfl/beatlink/ui/library/EditPlaylist.kt:227]: The new image is uploaded inside of the `permissionLauncher` on line 91

**Review by lunasrc — APPROVED** [review]
Thank you for the changes and clarification.
This PR is approved !

**Pull request merged**